### PR TITLE
Feat/report error

### PR DESCRIPTION
**Pull request by xitronix**
Closes https://github.com/arcxmoney/analytics-sdk/issues/54
Blocked by https://github.com/arcxmoney/monorepo/pull/2057

### CODE REVIEW

File: src/ArcxAnalyticsSdk.ts [hunk 1]
```diff
@@ -184,13 +185,17 @@ export class ArcxAnalyticsSdk {
 
   private async _getCurrentChainId(): Promise<string> {
     if (!window.ethereum) {
-      throw new Error('ArcxAnalyticsSdk::_getCurrentChainId: No ethereum provider found')
+      const errorMsg = 'ArcxAnalyticsSdk::_getCurrentChainId: No ethereum provider found'
+      this._reportError(errorMsg)
+      throw new Error(errorMsg)
```
Comment (gtupak) [inline src/ArcxAnalyticsSdk.ts:190]: I'd make a pure utility function `reportError(sdk:ArcxAnalyticsSdk, error: any)` that posts the error then throws. This way we can use it everywhere in any file and save some additional lines

**Review by gtupak — CHANGES_REQUESTED** [review]

File: src/ArcxAnalyticsSdk.ts (hunk 1 above)
Comment (xitronix) [reply src/ArcxAnalyticsSdk.ts]: `reportError` is private so i extracted `_reportErrorAndThrow`

File: src/ArcxAnalyticsSdk.ts [hunk 2]
```diff
@@ -253,6 +255,20 @@ export class ArcxAnalyticsSdk {
     return true
   }
 
+  /** Report error to the server in order to better understand edge cases which can appear */
+  _reportError(error: string): Promise<string> {
+    return postRequest(this.sdkConfig.url, this.apiKey, '/report-error', {
+      identityId: this.identityId,
+      error,
+    })
```
Comment (gtupak) [inline src/ArcxAnalyticsSdk.ts:263]: This depends on https://github.com/arcxmoney/monorepo/pull/2057. Will merge for now but just as an FYI

**Review by gtupak — APPROVED** [review]

File: src/ArcxAnalyticsSdk.ts (hunk 2 above)
Comment (xitronix) [reply src/ArcxAnalyticsSdk.ts:263]: Agree, I mentioned it in pr description

**Pull request merged**